docs/src/main/asciidoc/security-openid-connect-dev-services.adoc

```diff
@@ -0,0 +1,139 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Quarkus - Dev Services for OpenId Connecta
+
+include::./attributes.adoc[]
+
+This guide covers the Dev Services for OpenId Connect Keycloak provider and explains how to support other OpenId Connect providers.
+
+== Introduction
+
+Quarkus now ships with link:dev-ui[Dev UI] which is available in dev mode (when you start quarkus with `mvn quarkus:dev`) at http://localhost:8080/q/dev[/q/dev] by default.
+
+When you start your application with only a `quarkus-oidc` dependency:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-oidc</artifactId>
+</dependency>
+----
+
+then the main Dev UI page will include an `OpenId Connect Card`:
+
+image::dev-ui-oidc-card.png[alt=Dev UI OpenId Connect Card,role="center"]
+
+Next please add a Dev Services extension supporting OpenId Connect Provider you work with.
+
+=== Keycloak
+
+Add the following dependency:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-devservices-keycloak</artifactId>
+</dependency>
+----
+
+and start your application, without configuring `quarkus.oidc` properties in `application.properties`. You will see in the console something similar to:
+
+[source,shell]
+----
+$ mvn quarkus:dev
+
+2021-06-04 16:22:47,175 INFO  [🐳 .io/keycloak/keycloak:latest]] (build-38) Creating container for image: quay.io/keycloak/keycloak:latest
+2021-06-04 16:22:47,243 INFO  [🐳 .io/keycloak/keycloak:latest]] (build-38) Starting container with ID: 6469f6db9cec2c855fcc6c8db4273944cc9d69e8f6803a0b47eb2d5b8f5b94fd
+2021-06-04 16:22:47,629 INFO  [🐳 .io/keycloak/keycloak:latest]] (build-38) Container quay.io/keycloak/keycloak:latest is starting: 6469f6db9cec2c855fcc6c8db4273944cc9d69e8f6803a0b47eb2d5b8f5b94fd
+2021-06-04 16:22:47,643 INFO  [org.tes.con.wai.str.HttpWaitStrategy] (build-38) /elastic_lovelace: Waiting for 60 seconds for URL: http://localhost:32812/auth (where port 32812 maps to container port 8080)
+2021-06-04 16:23:07,665 INFO  [🐳 .io/keycloak/keycloak:latest]] (build-38) Container quay.io/keycloak/keycloak:latest started in PT5.500489S
+...
+2021-06-04 16:23:11,155 INFO  [io.quarkus] (Quarkus Main Thread) security-openid-connect-quickstart 1.0.0-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 25.968s. Listening on: http://localhost:8080
+2021-06-04 16:23:11,157 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
+----
+
+The `quay.io/keycloak/keycloak:latest` Keycloak image is used to start a container. The `quay.io/keycloak/keycloak` image version can be changed by setting a `keycloak.version` property.
```

Comment:
We can't use the `:latest` version, as it results in un-reproducable builds/tests. You don't want you CI to suddenly break because a new version of Keycloak has been released.